crypto/ml_kem/ml_kem.c

```diff
@@ -167,12 +169,92 @@ static void scalar_encode(uint8_t *out, const scalar *s, int bits);
 
 #endif
 
+/*
+ * Indices of slots in the vinfo tables below
+ */
+#define ML_KEM_512_VINFO    0
+#define ML_KEM_768_VINFO    1
+#define ML_KEM_1024_VINFO   2
+
+/*-
+ * Tables describing supported ASN.1 input/output formats.
+ *
+ * On output the PKCS8 info table order is important:
+ * - When we have a seed we'll use the first entry with a non-zero seed offset.
+ * - Otherwise, the first entry with a zero seed offset.
+ *
+ * As written, when possible, we prefer to output both the seed and private
+ * key, otherwise, just the private key ([1] IMPLICIT OCTET STRING form).
+ *
+ * The various lengths in the PKCS#8 tag/len fields could have been left
+ * zeroed, and filled in on the fly from the algorithm parameters, but that
+ * makes the code more complex, so a choice was made to embed them directly
+ * into the tables.  Had they been zeroed, one table could cover all three
+ * algorithms.
+ */
+
+/*-
+ * ML-KEM-512:
+ * Public key bytes:   800 (0x0320)
+ * Private key bytes: 1632 (0x0660)
+ */
+static const ML_KEM_SPKI_INFO ml_kem_512_spki_info = {
+    { 0x30, 0x82, 0x03, 0x32, 0x30, 0x0b, 0x06, 0x09, 0x60, 0x86, 0x48,
+      0x01, 0x65, 0x03, 0x04, 0x04, 0x01, 0x03, 0x82, 0x03, 0x21, 0x00, }
+};
+static const ML_KEM_PKCS8_INFO ml_kem_512_pkcs8_info[] = {
+    { "SEED_PRIV", 1706, 0x308206a6, 0x0440, 6, 0x81820660, 74, 0, },
```

Comment:
The format can have optional attributes at the end..
Does this handle this?
      PrivateKeyInfo ::= SEQUENCE {
        version                   Version,
        privateKeyAlgorithm       PrivateKeyAlgorithmIdentifier,
        privateKey                PrivateKey,
        attributes           [0]  IMPLICIT Attributes OPTIONAL }

Reply:
Not a problem, those attributes are not part of the `PKCS#8` `PrivateKey` OCTET STRING, which is all we're encoding or decoding.